File: api/src/shared/deal/Deal.ts
import { DayOfWeek } from "../DayOfWeek"
import { DealFeature } from "./DealFeature"

export enum DealStatus {
    new = "new", // Needs verification
    normal = "normal",
    expired = "expired",
    invalid = "invalid" 
}

export interface Deal {
    
    status: DealStatus
    compoundID: string
    days: DayOfWeek[]
    features: DealFeature[]
    text: string
    link?: string
    created: Date
    timeStart: number // Minutes from 00:00
    timeEnd: number
    
}

File: ios/LocalDeals/Model/Deal.swift
//  Created by Alexander Skorulis on 1/7/2022.

import Foundation

struct Deal: Codable {
    let timeStart: Int?
    let timeEnd: Int?
    let days: [Int]?
    let features: [Int]
    let placeID: String
    let link: String?
    let text: String
    let compoundID: String
    let status: String
    
}

extension Deal: Identifiable {
    var id: String { compoundID }
}
